### DISCUSSION TITLE
Error when trying to query `auth.users` related data

### DISCUSSION THREAD

**Discussion by mcewen87** [supabase/supabase#2947] [Questions] [answered]
# Bug report

## Describe the bug

When trying to make a select query using the Supabase client that joins a related table, I'm receiving the following error: 
"Could not find a relationship between posts and users in the schema cache"

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create a posts table with a fk constraint 'user_id' on the auth.users table.
2. Attempt to query all posts by user id and join the email field from auth.users as such

```
  const { data, error } = await supabase
      .from("posts")
      .select(
        `id,
        title,
        description,
        users (
          email
        )
        `
      )
      .eq("user_id", user.id);

```
This throws an error: "Could not find a relationship between posts and users in the schema cache"

## Expected behavior

I expect email to return from the auth.users table. 

## Additional context

In the Supabase dashboard, I can see a fk relationship exists on the posts table for user_id, I can also see the constraint listed from pg_constraints. 
![Screen Shot 2021-08-22 at 8 47 26 PM](https://user-images.githubusercontent.com/23143729/130379232-ddeeb259-3e03-4e18-b0d8-009f7808bb6a.png)
![Screen Shot 2021-08-22 at 8 46 57 PM](https://user-images.githubusercontent.com/23143729/130379245-69b9ee90-272e-4e33-9d9d-6aabb0de8000.png)

**Comment by steve-chavez** [comment, marked as answer]
> Create a posts table with a fk constraint 'user_id' on the auth.users table.

**Edit**: Make sure to create the adequate GRANT and RLS POLICY on auth.users when exposing it in a view like this

The `auth.users` table is part of the `auth` schema and the `supabase-js` library only works on objects inside the `public` schema.

What you could do is to create a view of `auth.users` on the `public` schema:

```sql
create view users as select id, email from auth.users;
-- IMPORTANT: only expose a subset of auth.users columns in the view, it has things like encrypted_password, which should never be public
```

Then querying related data should work:

```js
  const { data, error } = await supabase
      .from("posts")
      .select(
        `id,
        title,
        description,
        users (
          email
        )
        `
      )
      .eq("user_id", user.id);
```

**Reply by AJamesPhillips** [reply]
@steve-chavez is it correct that an anonymous user visiting your site could get a list of all the users email addresses?

**Reply by bnjmnt4n** [reply]
@AJamesPhillips I tried doing something similar in a project, and indeed creating a view like that exposes user information of all selected columns from `auth.users`. Instead, my workaround was to create a new table referencing `auth.users` with RLS policies to ensure users can only see their user information, or user information of someone in the same team for example, then add triggers on `auth.users` to ensure that the relevant columns will be kept up to date.
